src/components/VerificationTracker/AddAddress.js

```diff
@@ -1,34 +1,75 @@
-import React, { useContext } from "react";
-import { mutate } from "swr";
+import React, { useContext, useState } from "react";
 import AddressForm from "components/addresses/AddressForm";
 import gpib from "apis/gpib";
 import { AuthContext } from "components/auth/Auth";
+import useSWR, { mutate } from "swr";
+import ErrorMessage from "components/ErrorMessage";
+import Loader from "components/Loader";
+import { Button, Spinner } from "react-bootstrap";
 
 const AddAddress = () => {
   const { user } = useContext(AuthContext);
-
+  const { data: userHDAddress, error: fetchHDAddressError } = useSWR(
+    `/user/${user.id}/address`
+  );
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const isFetchingHDAddress = !userHDAddress && !fetchHDAddressError;
   const addAddress = async (v, actions) => {
     const parsedValues = { ...v, userID: user?.id, percent: Number(v.percent) };
     try {
       await gpib.secure.post(`/address`, parsedValues);
       actions.setSubmitting(false);
+      await mutate(`/user/${user.id}/address`);
     } catch (e) {
       console.log(e);
       actions.setErrors({ hidden: e });
       actions.setSubmitting(false);
     }
   };
+
+  const generateHDAddress = async () => {
+    setIsSubmitting(true);
+    try {
+      await gpib.secure.post("/User/HDAddress");
+      await mutate(`/user/${user.id}/address`);
+      setIsSubmitting(false);
+    } catch (error) {
+      console.error(error);
+      setIsSubmitting(false);
```

Comment:
I would put the setIsSumbitting(false) inside a finally rather than in both the try and the catch but I guess that is just personal preference.